scalafix-core/shared/src/main/scala/scalafix/v1/SemanticDoc.scala

```diff
@@ -0,0 +1,144 @@
+package scalafix.v1
+
+import java.nio.file.Files
+import java.util
+import scala.meta.io.Classpath
+import scala.meta.io.RelativePath
+import scala.collection.mutable.ListBuffer
+import scala.meta._
+import scala.meta.contrib.AssociatedComments
+import scalafix.internal.util.SymbolTable
+import scalafix.util.MatchingParens
+import scalafix.util.TokenList
+import scala.meta.internal.{semanticdb3 => s}
+import scalafix.internal.patch.DeprecatedSemanticdbIndex
+import scalafix.internal.v1._
+import scalafix.util.SemanticdbIndex
+
+trait SemanticContext
+
+final class SemanticDoc private[scalafix] (
+    val doc: Doc,
+    // privates
+    private[scalafix] val sdoc: s.TextDocument,
+    private[scalafix] val symtab: SymbolTable
+) extends SemanticContext {
+
+  override def toString: String = s"SemanticDoc(${input.syntax})"
+  def toLegacy: SemanticdbIndex = new DeprecatedSemanticdbIndex(this)
+
+  // =============
+  // Syntactic API
+  // =============
+  def tree: Tree = doc.tree
+  def tokens: Tokens = doc.tokens
+  def input: Input = doc.input
+  def matchingParens: MatchingParens = doc.matchingParens
+  def tokenList: TokenList = doc.tokenList
+  def comments: AssociatedComments = doc.comments
+
+  // ============
+  // Semantic API
+  // ============
+  def symbol(tree: Tree): Sym = {
+    val result = symbols(TreePos.symbol(tree))
+    if (!result.hasNext) Sym.None
+    else result.next() // Discard multi symbols
+  }
+
+  def info(tree: Tree): Sym.Info = info(symbol(tree))
```

Comment:
How much do we need this method? Strictly speaking, a tree doesn't have an info, so it was always a bit strange to me that there is a method like that. In comparison, in scalac, every tree does have a tpe, so `Tree.tpe` is appropriate, imo.

Reply:
I agree, this method should not exist. Removed

(PS. because no code is written against v1 yet we can tweak it to our desires 😄 )